### PR TITLE
Add CSSTree to parsers

### PR DESCRIPTION
**Pull request by lahmatiy**
Added CSSTree – detailed CSS parser (https://github.com/csstree/csstree)

### CODE REVIEW

File: src/parsers/css/csstree.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  version: pkg.version,
+  homepage: pkg.homepage || 'https://github.com/csstree/csstree',
+  locationProps: new Set(['loc']),
+
+  loadParser(callback) {
+    require(['css-tree'], callback);
+  },
+
+  parse(csstree, code) {
+    return JSON.parse(JSON.stringify(csstree.parse(code, {
```
Comment (fkling) [inline src/parsers/css/csstree.js]: Why is this needed here? Does `.parse` not return a plain object?

Comment (lahmatiy) [inline src/parsers/css/csstree.js]: It does, but CSSTree AST stores children nodes as a bi-directional list (a List instance), not as an array. If it were possible to make preprocessing of values, it would be easier:

```js
preprocessValue(value) {
    return value instanceof csstree.List ? value.toArray() : value;
}
```

Or we can convert `List` instances to array by walker:

```js
parse(csstree, code) {
    var ast = csstree.parse(code, {
        positions: true
    });
    csstree.walkUp(ast, function(node) {
        if (node.children && node.children instanceof csstree.List) {
            node.children = node.children.toArray();
        }
    });
    return ast;
}
```
It's more verbose, but if you prefer that way I can update the PR ;)
Maybe there will be a method to make it easier out of the box. But I'm still thinking about it.

Comment (lahmatiy) [inline src/parsers/css/csstree.js]: Updated `csstree` to new version with `toPlainObject()` method on board. Now it looks better ;)

**Comment by lahmatiy** [comment]
@fkling Is there a chance for merge?

**Comment by fkling** [comment]
Yeah, sorry for the delay. I'm currently focusing on a big change ( #184 ). I will merge your changes on top of that and publish a new version either today or tomorrow.

**Comment by lahmatiy** [comment]
@fkling Cool! Thank you!

**Comment by fkling** [comment]
Sorry, I didn't find much time to work on astexplorer the last two days. I'll definitely get to it this week though.

**Comment by lahmatiy** [comment]
It's ok. Thank you

**Comment by fkling** [comment]
I rebased and merged this PR in 87bc4f879b8882399517a09420ad5cd9f63ffabf . Thank you!

**Comment by lahmatiy** [comment]
Awesome! Thank you so much 👍
